rollup.config.js

```diff
@@ -37,7 +39,13 @@ function createDeclarationConfig(input, output) {
       dir: output,
     },
     external,
-    plugins: [typescript({ declaration: true, outDir: output })],
+    plugins: [
+      typescript({
+        declaration: true,
+        emitDeclarationOnly: true,
```

Comment:
What happens without this?

Reply:
emitDeclarationOnly: true => only creates `src/index.d.ts` and other `.d.ts` files, if you remove that, an additional `index.js` is also created, which overrides the `index.js` generated by rollup, thus creating disparity into the expected output and actual output of the rollup config.

Reply:
wow, really. so, it must have been a long-existed issue.

Reply:
Maybe, I observed it to be happening during the initial ESM PR

Reply:
I confirmed. The js files created by tsc seem overwritten by files by rollup. So, there has been no issues. It's nice to have, though.